### ISSUE TITLE
Feature: add headers

### ISSUE THREAD

**Issue by TomFreudenberg** [4commerce-technologies-AG/midi-smtp-server#21]
Create a standard method to add MidiSmtpServer delivery information to headers.

Check also for some useful X-Headers.

**Comment by gencer** [comment]
Hi @TomFreudenberg, This is really a good feature.

At the moment I do this on `on_message_data_event`. I manually create a MIME object and apply all necessary headers.

If I'm not wrong, "Probably" you will add a hook around this event.

**Comment by TomFreudenberg** [comment]
Hi @gencer

you still will be able to do like you do.

I would like to append headers directly to the list of headers. So I think about 1 or 2 new events like:

```ruby
# event when beginning of message DATA
def on_message_data_initialize_event(ctx)
  # check whatever you like and append additional header data 
  ctx[:message][:data] << "Received: by me\r\n"
  # or use (new) standard method like
  ctx[:message][:data] << local_received_header(Time.now)
end
```

```ruby
# event when headers where sent while receiving message DATA
def on_message_data_headers_sent_event(ctx)
  # check and append all headers
  ctx[:message][:data] << "X-myheader: Super information\r\n"
end
```

Cheers
Tom

**Comment by TomFreudenberg** [comment]
Hi @gencer 

feature is implemented in release 2.3.1.

Please check documentation at: https://github.com/4commerce-technologies-AG/midi-smtp-server#adding-and-testing-headers

Should be what you are looking for.

Cheers
Tom

**Comment by TomFreudenberg** [comment]
Thanks @gencer
